cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go

```diff
@@ -462,6 +475,35 @@ func (o *ScaleUpOrchestrator) ComputeExpansionOption(
 		return option
 	}
 
+	autoscalingOptions, err := nodeGroup.GetOptions(o.autoscalingContext.NodeGroupDefaults)
+	if err != nil {
+		klog.Errorf("Failed to get autoscaling options for node group %s: %v", nodeGroup.Id(), err)
+	}
+	if autoscalingOptions != nil && autoscalingOptions.AtomicScaleUp {
+		// If atomic, there should be no similar node groups for balancing
+		// Also the number of pods should be capped according to the node count.
+
+		// Limit the number of nodes we estimate to the max scale up possible.
+		currentTargetSize, err := nodeGroup.TargetSize()
+		if err != nil {
+			klog.Errorf("Failed to get node group size: %v", err)
+			return option
+		}
+		// When the node group has the atomic scale up option, the number of nodes
+		// that can be added to the node group should be capped according to the
+		// maximum size of the node group and not the 'MaxNodesPerScaleUp' option.
+		// Since we cannot change or update the limiter when we build the estimator
+		// with the 'autoscalingContext.EstimatorBuilder' we create a new estimator
+		// with a different limiter here.
+		nodeEstimationCap := nodeGroup.MaxSize() - currentTargetSize
+		limiter := estimator.NewThresholdBasedEstimationLimiter(nodeEstimationCap, 0)
```

Comment:
I think recreating the limiter here is a really BAD idea for multitude of reasons:
- Limiter is an interface. The fact that we currently use the `ThresholdBasedEstimationLimiter` does not mean that we won't change it in the future and that this new implementation won't implement some *really* important functionality, which will be de-facto broken when used with AtomicScaleUp.
- Current `ThresholdBasedEstimationLimiter` was implemented to improve the scalability of the Cluster Autoscaler. Without it we had issues with large scale-ups which were causing the CA component to get stuck and fail healthchecks, resulting in crushlooping state. I'm *strongly* against avoiding these limits here, as this can result in crashlooping state.

But most importantly - why do you need this? I get that it would be "nice" to fit as many pods as you can, but after running the estimation you are still bumping the `NodeCount` to max possible. So it doesn't really make a difference what is the limit here. As I can see it with this change there are two posibilites:
- You are able to pack N nodes (N <= max nodes), and then change N to max nodes either way.
- You pack so many nodes that the CA becomes unhealthy and crashloops.

I don't see any reason to change the limiter here - you want to scale to max no matter what, so who cares when the limiter stops the binpacking? They only effect is that you need a hacky and hard to maintain piece of code, that *also* can result in the component crashlooping. Nothing other than that changes.

Reply:
As we have discussed offline, it turns out we don't need to create a new estimator and using existing one works out for us. Removed this part, now we just set the node count after the estimation is performed for atomic scale up.